.github/workflows/build.yaml

```
@@ -0,0 +1,58 @@
+name: push
+on:
+  push:
+  pull_request:
+jobs:
+  test:
+    strategy:
+      matrix:
+        os: [ubuntu-latest]
+        python-version: ["3.7", "3.8", "3.9", "3.10", "3.11"]
+        # we do not want a large number of windows and macos builds, so
+        # enumerate them explicitly
+        include:
+          - os: windows-latest
+            python-version: 3.9
+          - os: macos-latest
+            python-version: 3.9
+          - os: ubuntu-20.04
+            python-version: 3.6
+    name: "test py${{ matrix.python-version }} on ${{ matrix.os }} "
```

I prefer more readable test names. Below, my suggestion adds an `os-name` (and removes trailing whitespace in the test name), resulting in test names like "Test py3.11 on Windows".

```suggestion
        os: [ubuntu-latest]
        os-name: [Ubuntu]
        python-version: ["3.7", "3.8", "3.9", "3.10", "3.11"]
        # we do not want a large number of windows and macos builds, so
        # enumerate them explicitly
        include:
          - os: windows-latest
            os-name: Windows
            python-version: 3.9
          - os: macos-latest
            os-name: MacOS
            python-version: 3.9
          - os: ubuntu-20.04
            os-name: Ubuntu 20.04
            python-version: 3.6
    name: "Test py${{ matrix.python-version }} on ${{ matrix.os-name }}"
```

---

I'm somewhat agnostic towards this change, but I understand why you prefer it. I generally wouldn't bother doing it myself unless pressed, but I more than happily accept it if someone else does it! 😁 

Also, good catch on the trailing space I had in there!

Thanks for the suggestion; applying it now.